salt/returners/mongo_future_return.py

```diff
@@ -178,17 +178,15 @@ def _get_conn(ret):
         mdb = conn.get_database()
     else:
         if PYMONGO_VERSION > _LooseVersion("2.3"):
-            conn = pymongo.MongoClient(host, port)
+            conn = pymongo.MongoClient(host, port, username=user, password=password)
```

Comment:
Do we need the username and password for the above client declarations? Line 177 here.

Reply:
Yes, otherwise pymongo throws the following exception:

`pymongo.errors.OperationFailure: command insert requires authentication, full error: {'ok': 0.0, 'errmsg': 'command insert requires authentication', 'code': 13, 'codeName': 'Unauthorized'}`